lib/vagrant-service-manager/service.rb

```diff
@@ -0,0 +1,34 @@
+Dir["#{File.dirname(__FILE__)}/services/*.rb"].each { |f| require_relative f }
+
+module Vagrant
+  module ServiceManager
+    SUPPORTED_BOXES = ['adb', 'cdk']
+
+    class Service
+      def initialize(app, env)
+        @app = app
+        @machine = env[:machine]
+        @ui = env[:ui]
+        @services = @machine.config.servicemanager.services.split(',').map(&:chomp)
+        @docker_hook = Docker.new(@machine, @ui)
+        @openshift_hook = OpenShift.new(@machine, @ui)
+      end
+
+      def call(env)
+        if SUPPORTED_BOXES.include? @machine.guest.capability(:flavor)
+          @docker_hook.execute
```

Comment:
It seems that there should be a check here.  I should be allowed to say I don't want Docker (or later OpenShift).